### PR TITLE
format with nixpkgs-fmt

### PR DESCRIPTION
**Pull request by zimbatm**
What do you think of the format @nmattia ?

### CODE REVIEW

**Comment by nmattia** [comment]
I like it! The latest changes actually are in https://github.com/nmattia/naersk/tree/nm-cargo-release, let me merge that for you first.

**Comment by nmattia** [comment]
https://github.com/nmattia/naersk/pull/16

File: builtins/default.nix [hunk 1]
```diff
@@ -7,54 +7,57 @@
...
+  pathExists = path:
+    let
+      all = lib.all (x: x);
+      isOk = part:
+        let
+          dir = builtins.dirOf part;
+          basename = builtins.unsafeDiscardStringContext (builtins.baseNameOf part);
+          dirContent = builtins.readDir dir;
+        in
+          builtins.hasAttr basename dirContent && # XXX: this may not work if the directory is a symlink
```
Comment (nmattia) [inline builtins/default.nix]: @zimbatm looks like this guy got moved! Issue when formatting infix operators?

File: builtins/to-toml.nix [hunk 2]
```diff
@@ -25,106 +25,115 @@ let
...
-    if typeOf x == "set" then
-      if lib.isDerivation x then "string" else "set" else
-    if typeOf x == "list" then
-      if length x == 0 then "list"
-      else
-        let ty = typeOf (elemAt x 0); in
-        #assert (all (v: typeOf v == ty) x);
-        if ty == "set" then "list_of_attrs" else "list"
-    else abort "Not implemented: toml type for ${typeOf x}";
+      if typeOf x == "bool" then "bool" else
```
Comment (nmattia) [inline builtins/to-toml.nix]: mh this is a bit annoying, it gets much harder to read

Comment (nmattia) [reply builtins/to-toml.nix]: according to some of the formatting later on in this module, putting the `else` on the next line should fix it

File: builtins/default.nix (hunk 1 above)
Comment (zimbatm) [reply builtins/default.nix]: hmm maybe. I'll just fix it manually for now

File: builtins/to-toml.nix (hunk 2 above)
Comment (zimbatm) [reply builtins/to-toml.nix]: what do you think of the new format?

File: builtins/default.nix (hunk 1 above)
Comment (zimbatm) [reply builtins/default.nix]: moved the code around a bit

**Comment by zimbatm** [comment]
@nmattia pushed an update with the latest greatest version of nixpkgs-fmt. I skipped nix/sources.nix

**Comment by nmattia** [comment]
Looks great, thanks @zimbatm !

**Pull request merged**